babi/buf.py

```diff
@@ -299,6 +299,43 @@ def left(self, margin: Margin) -> None:
         else:
             self.x -= 1
 
+    def paragraph_down(self, margin: Margin) -> None:
```

Comment:
I think this should be implemented in `File` and not in `buf`

you also don't need `set_x_after_vertical_movement` because you're always going to set x to 0